solver/src/main.rs

```diff
@@ -211,9 +202,16 @@ async fn main() {
     let registry = Registry::default();
     let metrics = Arc::new(Metrics::new(&registry).expect("Couldn't register metrics"));
 
-    // TODO: custom transport that allows setting timeout
-    let transport =
-        create_instrumented_transport(HttpTransport::new(args.shared.node_url), metrics.clone());
+    let client = reqwest::ClientBuilder::new()
+        .timeout(args.shared.http_timeout)
+        .user_agent("gp-v2-services/2.0.0")
+        .build()
+        .unwrap();
```

Comment:
nit: maybe move these 4 lines into shared so that we have the same config between API and solver until we need changes?

Reply:
changed